src/prefect/client/orion.py

```diff
@@ -152,6 +149,29 @@ def __init__(
                 )
             httpx_settings.setdefault("base_url", api)
 
+            # See https://www.python-httpx.org/advanced/#pool-limit-configuration
+            httpx_settings.setdefault(
+                "limits",
+                httpx.Limits(
+                    # We see instability when allowing the client to open many connections at once.
+                    # Limiting concurrency results in more stable performance.
+                    max_connections=16,
+                    max_keepalive_connections=8,
+                    # The Prefect Cloud LB will keep connections alive for 30s.
+                    # Only allow the client to keep connections alive for 25s.
+                    keepalive_expiry=25,
+                ),
+            )
+            # See https://www.python-httpx.org/advanced/#custom-transports
+            # `retries` specifies the number of retries on connection errors
+            httpx_settings.setdefault("transport", httpx.AsyncHTTPTransport(retries=3))
```

Comment:
it seems to me that this prevents that the other transport related parameters such as connection limits and http2 settings are properly applied.
httpx only creates a new transport object if no such object is provided as a parameter (see [https://github.com/encode/httpx/blob/master/httpx/_client.py#L1439](https://github.com/encode/httpx/blob/master/httpx/_client.py#L1439)). 
i don't see how one could manipulate the retry behavior other than via the transport parameter. so this could be possibly fixed by moving the limits and http2 setting definition into the transport constructor.